exercicios/para-casa/emilly-santos/entrega.js

```diff
@@ -0,0 +1,68 @@
+function Cat(name, age, color, castrated) {
+    this.name = name;
+    this.age = age;
+    this.color = color;
+    this.castrated = castrated
+    this.historic = new MedicalHistory();
+}
+
+Cat.prototype.meow = function() {
+    console.log(`O gato ${this.nome} está miando`)
+}
+
+function Dog(name, age, color, castrated, race) {
+    this.name = name;
+    this.agr = age;
+    this.color = color;
+    this.castrated = castrated;
+    this.race = race;
+    this.historic = new MedicalHistory();
+}
+
+Dog.prototype.bark = function() {
+    console.log(`O gato ${this.nome} está latindo`)
+}
+
+function ExoticAnimal(name, age, color, species, adult) {
+    this.name = name;
+    this.age = age;
+    this.color = color;
+    this.species = species;
+    this.adult = adult;
+    this.historic = new MedicalHistory();
+}
+
+ExoticAnimal.prototype.ehAdulto = function(){
+    if(this.species === 'hamster' && this.age >=2){
+        return true;
+    }else if(this.species ==='papagaio' && this.age >=36){
+        return true;
+    }else if (this.species === 'calopsita' && this.age >=18){
+        return true;
+    }else{
+        return false;
+    }
+}
+
+function MedicalHistory() {
+    this.queries = [];
+    this.vaccines = []; 
+    this.procedures = [];
+}
+
+MedicalHistory.prototype.consult = function(date){
+    this.queries.push(date);
```

Comment:
Aqui o ideal seria usar o unshift para que as datas mais recentes fiquem antes das datas mais antigas.
Fica assim: "this.queries.unshift(date)".